db/helpers/botDataUtils.js

```diff
@@ -39,6 +39,30 @@ function isJson(item) {
   return false;
 } // isJson
 
+/**
+ * Bot Data values are stored as strings but can be converted to
+ * JSON. If either parameter is not JSON strings we replace originalValue
+ * with the newValue. If both values are JSON strings we extend the
+ * originalValue by adding any missing key-value pairs and replacing any
+ * exsiting key-value pairs with new value
+ *
+ * @param {String} originalValue - Original bot data value
+ * @param {String} newValue - New bot data value
+ * @returns {String} - Bot data where values are properly merged
+ */
+function combineValue(originalValue, newValue) {
```

Comment:
One of the big reasons to pull a function like this out to a separate module is that you get the opportunity to unit test the heck out of it separately from the context in which it will be called. Where are the tests?